gnomad_qc/v4/assessment/calculate_per_sample_stats.py

```diff
@@ -54,206 +62,364 @@
+
+    return map_filter_field_to_meta
+
+
+def get_capture_filter_exprs(
+    ht: hl.Table,
+    pass_filter_expr: Optional[hl.expr.BooleanExpression] = None,
+    ukb_capture: bool = False,
+    broad_capture: bool = False,
+) -> Dict[str, hl.expr.BooleanExpression]:
+    """
+    Get filter expressions for UK Biobank and Broad capture regions.
+
+    :param ht: Table containing variant annotations. The following annotations are
+        required: 'region_flags'.
+    :param pass_filter_expr: Expression for variants that pass all variant QC filters.
+    :param ukb_capture: Include count of variants that are in UKB capture intervals.
+    :param broad_capture: Include count of variants that are in Broad capture intervals.
+    :return: Dictionary of filter expressions for UK Biobank and Broad capture regions.
+    """
+    filter_expr = {}
+    log_list = []
+    if ukb_capture:
+        log_list.append("variants in UK Biobank capture regions")
+        filter_expr["capture_ukb"] = ~ht.region_flags.outside_ukb_capture_region
+
+    if broad_capture:
+        log_list.append("variants in Broad capture regions")
+        filter_expr["capture_broad"] = ~ht.region_flags.outside_broad_capture_region
+
+    if ukb_capture and broad_capture:
+        log_list.append("variants in the intersect of UKB and Broad capture regions")
+        filter_expr["capture_ukb_broad_intersect"] = (
+            filter_expr["capture_ukb"] & filter_expr["capture_broad"]
+        )
+
+        log_list.append("variants in the union of UKB and Broad capture regions")
+        filter_expr["capture_ukb_broad_union"] = (
+            filter_expr["capture_ukb"] | filter_expr["capture_broad"]
+        )
+
+    # Include capture filters for variants that pass all variant QC filters.
+    if pass_filter_expr is not None:
+        log_list.append("All of the above for only PASS variants")
+        add_pass_expr = {
+            f"variant_qc_pass_{k}": v & pass_filter_expr for k, v in filter_expr.items()
+        }
+        filter_expr.update(add_pass_expr)
+
+    logger.info("Adding filtering for:\n\t%s...", "\n\t".join(log_list))
+
+    return filter_expr
+
+
+def get_summary_stats_filter_groups_ht(
+    ht: hl.Table,
     pass_filters: bool = False,
     ukb_capture: bool = False,
     broad_capture: bool = False,
     by_csqs: bool = False,
-    rare_variants: bool = False,
     vep_canonical: bool = True,
     vep_mane: bool = False,
     rare_variants_afs: Optional[list[float]] = None,
 ) -> hl.Table:
     """
-    Create Table of Hail's sample_qc output broken down by requested variant groupings.
-
-    Useful for finding the number of variants per sample, either all variants, or
-    variants fall into specific capture regions, or variants that are rare
-    (adj AF <0.1%), or variants categorized by predicted consequences in all, canonical
-    or mane transcripts.
+    Create Table of filter groups for summary stats.
```

Comment:
Very understated, think this needs a lot more depth. 

'Create Tables annotated with an array of booleans for which filter groups it falls into or not.' 

might be more descriptive

Reply:
sounds good, please add as a suggestion in the future to make it easier for me to add